server/app/services/export/CsvColumnFactory.java

```diff
@@ -0,0 +1,490 @@
+                        .orElse(""))
+            .build());
+  }
+
+  private Stream<Column> buildColumnsForCurrencyQuestion(
+      CurrencyQuestion q, ColumnType columnType) {
+    return Stream.of(
+        Column.builder()
+            .setColumnType(columnType)
+            .setHeader(formatHeader(q.getCurrencyPath()))
+            .setQuestionPath(q.getContextualizedPath())
+            .setAnswerExtractor(
+                cq ->
+                    ((CurrencyQuestion) cq)
+                        .getCurrencyValue()
+                        .map(Currency::getDollarsString)
+                        .orElse(""))
+            .build());
+  }
+
+  private Stream<Column> buildColumnsForDateQuestion(DateQuestion q, ColumnType columnType) {
+    return Stream.of(
+        Column.builder()
+            .setColumnType(columnType)
+            .setHeader(formatHeader(q.getDatePath()))
+            .setQuestionPath(q.getContextualizedPath())
+            .setAnswerExtractor(
+                dq ->
+                    ((DateQuestion) dq)
+                        .getDateValue()
+                        .map(
+                            localDate ->
+                                localDate.format(DateTimeFormatter.ofPattern("MM/dd/yyyy")))
+                        .orElse(""))
+            .build());
+  }
+
+  private Stream<Column> buildColumnsForEmailQuestion(EmailQuestion q, ColumnType columnType) {
+    return Stream.of(
+        Column.builder()
+            .setColumnType(columnType)
+            .setHeader(formatHeader(q.getEmailPath()))
+            .setQuestionPath(q.getContextualizedPath())
+            .setAnswerExtractor(eq -> ((EmailQuestion) eq).getEmailValue().orElse(""))
+            .build());
+  }
+
+  private Stream<Column> buildColumnsForFileUploadQuestion(
+      FileUploadQuestion q, ColumnType columnType, boolean isMultipleFileUploadEnabled) {
+    String baseUrl = settingsManifest.getBaseUrl().orElse("");
+
+    Column fileKeyListColumn =
+        Column.builder()
+            .setColumnType(columnType)
+            .setHeader(formatHeader(q.getFileKeyListPath()))
+            .setQuestionPath(q.getContextualizedPath())
+            .setAnswerExtractor(
+                fuq -> {
+                  FileUploadQuestion innerQ = (FileUploadQuestion) fuq;
+
+                  Stream<String> keyStream =
+                      isMultipleFileUploadEnabled
+                          ? innerQ
+                              .getFileKeyListValue()
+                              .map(ImmutableList::stream)
+                              .orElseGet(Stream::empty)
+                          : innerQ.getFileKeyValue().stream();
+
+                  return keyStream
+                      .map(fileKey -> ExportFormatUtils.formatFileUrlForAdmin(baseUrl, fileKey))
+                      .collect(Collectors.joining(", "));
+                })
+            .build();
+
+    if (isMultipleFileUploadEnabled) {
```

Comment:
This differs from the logic in getScalarAnswers() in that we use the feature flag to decide what columns to include. Before, we assumed from the presence of a `file_keys` field that the feature flag was on and data had been migrated.